libs/runtime/ebpf_hash_table.c

```diff
@@ -154,6 +156,33 @@ _ebpf_murmur3_32(_In_reads_((length_in_bits + 7) / 8) const uint8_t* key, size_t
     return hash;
 }
 
+static unsigned long
+_ebpf_compute_crc32(_In_reads_(length_in_bytes) const uint8_t* key, size_t length_in_bytes, uint32_t seed)
+{
+    // First process 8 bytes at a time.
+    uint32_t crc = seed;
+    uint8_t* start = (uint8_t*)key;
+    uint8_t* end = start + length_in_bytes;
+
+    while ((end - start) >= 8) {
+        crc = (uint32_t)_mm_crc32_u64(crc, *(uint64_t*)start);
+        start += 8;
+    }
+
+    // Process 4 bytes at a time.
+    while ((end - start) >= 4) {
```

Comment:
Approving though a possibly negligible perf improvement would be to change while to if, since this could only happen once.

Reply:
Thanks I will try that.

Reply:
Doesn't appear to make any detectable difference. Maybe the compiler is smart enough to deduece this already?